sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -381,4 +381,61 @@ public static Context mergeContexts(Context into, Context from) {
 
         return returnContext;
     }
+
+    /**
+     * Optimized version of {@link String#join(CharSequence, CharSequence...)} when the {@code values} has a small
+     * set of object.
+     *
+     * @param delimiter Delimiter between the values.
+     * @param values The values to join.
+     * @return The {@code values} joined delimited by the {@code delimiter}.
+     * @throws NullPointerException If {@code delimiter} or {@code values} is null.
+     */
+    public static String stringJoin(String delimiter, String... values) {
+        Objects.requireNonNull(delimiter, "'delimiter' cannot be null.");
+        Objects.requireNonNull(values, "'values' cannot be null.");
+
+        int count = values.length;
+        switch (count) {
+            case 0: return "";
+            case 1: return values[0];
+            case 2: return values[0] + delimiter + values[1];
+            case 3: return values[0] + delimiter + values[1] + delimiter + values[2];
+            case 4: return values[0] + delimiter + values[1] + delimiter + values[2] + delimiter + values[3];
+            case 5: return values[0] + delimiter + values[1] + delimiter + values[2] + delimiter + values[3]
+                + delimiter + values[4];
+            case 6: return values[0] + delimiter + values[1] + delimiter + values[2] + delimiter + values[3]
```

Comment:
Just curious, why we chose 6. Do we know if this is the sweet spot for string joins? Java seems to use 10 for such optimizations.

- [List.of()](https://docs.oracle.com/javase/9/docs/api/java/util/List.html#of-E-E-E-E-E-E-E-E-E-E-)
- [Map.of()](https://docs.oracle.com/javase/9/docs/api/java/util/Map.html#of-K-V-K-V-K-V-K-V-K-V-K-V-K-V-K-V-K-V-K-V-)

Reply:
I actually meant to get rid of this API as `String.join(CharSequence, CharSequence...)` doesn't have as much overhead as `String.join(CharSequence, Iterable<CharSequence>)` does and nothing uses this yet.